straxen/plugins/position_reconstruction.py

```diff
@@ -91,98 +109,84 @@ def compute(self, peaks):
+                                                            )
+        output = model.predict(area_per_channel_top)
 
         # writing output to the result
-        result['x_' + self.algorithm][peak_mask] = _out[:, 0]
-        result['y_' + self.algorithm][peak_mask] = _out[:, 1]
+        result['x_' + self.algorithm][peak_mask] = output[:, 0]
+        result['y_' + self.algorithm][peak_mask] = output[:, 1]
         return result
 
-    def _get_model_file_name(self):
-
-        config_file = f'{self.algorithm}_model'
-        model_from_config = self.config.get(config_file, 'No file')
-        if model_from_config == 'No file':
-            raise ValueError(f'{__class__.__name__} should have {config_file} '
-                             f'provided as an option.')
-        if isinstance(model_from_config, str) and os.path.exists(model_from_config):
-            # Allow direct path specification
-            return model_from_config
-        if model_from_config is None:
-            # Allow None to be specified (disables processing for given posrec)
-            return model_from_config
-
-        # Use CMT
-        model_file = straxen.get_correction_from_cmt(self.run_id, model_from_config)
-        return model_file
-
 
 @export
-@strax.takes_config(
-    strax.Option('mlp_model',
-                 help='Neural network model.' 
-                      'If CMT, specify as (mlp_model, ONLINE, True)'
-                      'Set to None to skip the computation of this plugin.',
-                 default=('mlp_model', "ONLINE", True), infer_type=False,
-                )
-)
 class PeakPositionsMLP(PeakPositionsBaseNT):
     """Multilayer Perceptron (MLP) neural net for position reconstruction"""
     provides = "peak_positions_mlp"
     algorithm = "mlp"
 
+    tf_model_mlp = straxen.URLConfig(
+        default=f'tf://download://cmt://{algorithm}_model?version=ONLINE&run_id=plugin.run_id',
+        help='MLP model. Should be opened using the "tf" descriptor. '
+             'Set to "None" to skip computation',
+        cache=True,
+    )
+
 
 @export
-@strax.takes_config(
-    strax.Option('gcn_model',
-                 help='Neural network model.' 
-                      'If CMT, specify as  (gcn_model, ONLINE, True)'
-                      'Set to None to skip the computation of this plugin.',
-                 default=('gcn_model', "ONLINE", True), infer_type=False,
-                )
-)
 class PeakPositionsGCN(PeakPositionsBaseNT):
     """Graph Convolutional Network (GCN) neural net for position reconstruction"""
     provides = "peak_positions_gcn"
     algorithm = "gcn"
     __version__ = '0.0.1'
 
+    tf_model_gcn = straxen.URLConfig(
+        default=f'tf://download://cmt://{algorithm}_model?version=ONLINE&run_id=plugin.run_id',
+        help='GCN model. Should be opened using the "tf" descriptor. '
+             'Set to "None" to skip computation',
+        cache=True,
+    )
+

```

Comment:
```suggestion
model = straxen.URLConfig(
        default=f'tf://download://cmt://format_arg://{algorithm}_model?version=ONLINE&run_id=plugin.run_id&algorithm=plugin.algorithm',
        help='TF model. Should be opened using a compatible protocol (e.g. tf://). '
             'Set to "None" to skip computation',
        cache=True,
    )

```

Reply:
Any reason youre not using the `format_arg://` to select which model name to load from CMT? I had added that protocol with this use case in mind...

Reply:
Could do, but to be honest I find it less readable than
```python
'tf://download://cmt://{algorithm}_model?version=ONLINE&run_id=plugin.run_id'
```


But again, only if I understand [your general comment correctly](https://github.com/XENONnT/straxen/pull/821#issuecomment-994839106)

Reply:
this comment is independent of the more general comment. Even if you have three different configs, you still need to format the string before its passed to cmt, otherwise cmt will just get the literal string `{algorithm}_model` as the name of the correction to load...

Reply:
@jmosbacher sorry, it is an f-string, so you get it from the class attribute (just sneakily changed my example 👿 )

Reply:
ah, very sneaky! ok great then this is fine :)